alibi/explainers/backends/cfrl_tabular.py

```diff
@@ -335,7 +335,7 @@ def sample_numerical(X_hat_num_split: List[np.ndarray],
 
 
 def sample_categorical(X_hat_cat_split: List[np.ndarray],
-                       C_cat_split: List[np.ndarray]) -> List[np.ndarray]:
+                       C_cat_split: Optional[List[np.ndarray]]) -> List[np.ndarray]:
```

Comment:
`None` case is dealt with in the body so it can clearly be `Optional` (as inferred by the type-checker elsewhere in the code).